app/src/main/java/com/duckduckgo/app/systemsearch/SystemSearchViewModel.kt

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2020 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.systemsearch
+
+import android.content.Intent
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.duckduckgo.app.autocomplete.api.AutoComplete
+import com.duckduckgo.app.autocomplete.api.AutoComplete.AutoCompleteResult
+import com.duckduckgo.app.global.DefaultDispatcherProvider
+import com.duckduckgo.app.global.DispatcherProvider
+import com.duckduckgo.app.global.SingleLiveEvent
+import com.jakewharton.rxrelay2.PublishRelay
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import timber.log.Timber
+import java.util.concurrent.TimeUnit
+
+class SystemSearchViewModel(
+    private val autoComplete: AutoComplete,
+    private val deviceAppLookup: DeviceAppLookup,
+    private val dispatchers: DispatcherProvider = DefaultDispatcherProvider()
+) : ViewModel() {
+
+    data class SystemSearchViewState(
+        val queryText: String = "",
+        val autocompleteResults: AutoCompleteResult = AutoCompleteResult("", emptyList()),
+        val appResults: List<DeviceApp> = emptyList()
+    )
+
+    sealed class Command {
+        object LaunchDuckDuckGo : Command()
+        data class LaunchBrowser(val query: String) : Command()
+        data class LaunchDeviceApplication(val intent: Intent) : Command()
+    }
+
+    val viewState: MutableLiveData<SystemSearchViewState> = MutableLiveData()
+    val command: SingleLiveEvent<Command> = SingleLiveEvent()
+
+    private val autoCompletePublishSubject = PublishRelay.create<String>()
+    private var autocompleteResults: AutoCompleteResult = AutoCompleteResult("", emptyList())
+
+    private var appsJob: Job? = null
+    private var appResults: List<DeviceApp> = emptyList()
+
+    init {
+        resetState()
+        configureAutoComplete()
+    }
+
+    private fun currentViewState(): SystemSearchViewState = viewState.value!!
+
+    fun resetState() {
+        autocompleteResults = AutoCompleteResult("", emptyList())
+        appsJob?.cancel()
+        appResults = emptyList()
+        viewState.value = SystemSearchViewState()
+    }
+
+    private fun configureAutoComplete() {
+        autoCompletePublishSubject
```

Comment:
We should probably handle the lint warning here about not holding onto a reference to this. Although all we do currently is update a `LiveData` - and that shouldn't matter currently if the `Activity` is already destroyed - it would be easy to accidentally do other things here and cause bugs.

Handling it is probably just a case of storing a reference to the `subscribe()` call as a `Disposable`, and disposing of that in `onCleared()`

     override fun onCleared() {
        super.onCleared()
        r?.dispose()
        r = null
    }